Redefinition errors for `shader-types-sharing` example in release build

As stated at https://github.com/vulkano-rs/vulkano/issues/1730#issuecomment-974712151, compiling the `shader-types-sharing` example in release mode results in a bunch of errors about redefined types. These errors don't appear in a debug build.

```
error[E0428]: the name `Data` is defined multiple times
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         `Data` redefined here
   |         previous definition of the type `Data` here
   |
   = note: `Data` must be defined only once in the type namespace of this module
   = note: this error originates in the macro `vulkano_shaders::shader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0428]: the name `SpecializationConstants` is defined multiple times
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         `SpecializationConstants` redefined here
   |         previous definition of the type `SpecializationConstants` here
   |
   = note: `SpecializationConstants` must be defined only once in the type namespace of this module
   = note: this error originates in the macro `vulkano_shaders::shader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0428]: the name `Parameters` is defined multiple times
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         `Parameters` redefined here
   |         previous definition of the type `Parameters` here
   |
   = note: `Parameters` must be defined only once in the type namespace of this module
   = note: this error originates in the macro `vulkano_shaders::shader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::clone::Clone` for type `main::shaders::ty::Parameters`
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         first implementation here
   |         conflicting implementation for `main::shaders::ty::Parameters`
   |
   = note: this error originates in the macro `vulkano_shaders::shader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::clone::Clone` for type `main::shaders::SpecializationConstants`
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         first implementation here
   |         conflicting implementation for `main::shaders::SpecializationConstants`
   |
   = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::default::Default` for type `main::shaders::SpecializationConstants`
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         first implementation here
   |         conflicting implementation for `main::shaders::SpecializationConstants`
   |
   = note: this error originates in the macro `vulkano_shaders::shader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::marker::Copy` for type `main::shaders::SpecializationConstants`
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         first implementation here
   |         conflicting implementation for `main::shaders::SpecializationConstants`
   |
   = note: this error originates in the derive macro `Copy` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::marker::Copy` for type `main::shaders::ty::Parameters`
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         first implementation here
   |         conflicting implementation for `main::shaders::ty::Parameters`
   |
   = note: this error originates in the derive macro `Copy` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::fmt::Debug` for type `main::shaders::SpecializationConstants`
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         first implementation here
   |         conflicting implementation for `main::shaders::SpecializationConstants`
   |
   = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `vulkano::shader::SpecializationConstants` for type `main::shaders::SpecializationConstants`
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         first implementation here
   |         conflicting implementation for `main::shaders::SpecializationConstants`
   |
   = note: this error originates in the macro `vulkano_shaders::shader` (in Nightly builds, run with -Z macro-backtrace for more info)
```

@Eliah-Lakhin this looks like it's related to the code that combines shader struct definitions, implemented in #1694.

--- a/vulkano-shaders/src/structs.rs
+++ b/vulkano-shaders/src/structs.rs
@@ -240,7 +240,7 @@ fn write_struct<'a>(
             return (quote! {}, total_size);
         }
 
-        debug_assert!(types_registry.insert(name, target_type).is_none());
+        assert!(types_registry.insert(name, target_type).is_none());
     }
 
     // We can only implement Clone if there's no unsized member in the struct.
@@ -804,7 +804,7 @@ pub(super) fn write_specialization_constants<'a>(
             return quote! {};
         }
 
-        debug_assert!(types_registry.insert(name, target_type).is_none());
+        assert!(types_registry.insert(name, target_type).is_none());
     }
 
     let map_entries = {

